core/dbt/parser/schema_yaml_readers.py

```diff
@@ -408,6 +423,79 @@ def __init__(self, schema_parser: SchemaParser, yaml: YamlBlock):
         self.schema_parser = schema_parser
         self.yaml = yaml
 
+    def _get_dimension_type_params(
```

Comment:
The code in these methods continues to be verbose and mechanical. We need a better way, though not today.

Reply:
I'm not sure I wholly agree 🤔 There are fields which will require remapping / conversion. To you're point though, a lot of the fields are just direct mappings. We could do something like `params = unparsed.to_dict()`, edit the dict to spell out the things that do indeed need to be changed, and then `SemanticModel(**params)`. However, this can hide some problems. For instance if a "direct mapping" has a typo'd property in either the node or the unparsed version, the key value will get dropped. Or if someone makes a change to the node (renaming or adding properties), it isn't immediately obvious that anything needs to be done in the `schema_yaml_parser.py` or `unparsed.py` (whereas this current implementation will start complaining pretty quick). 

Another way is we can push the mapping logic into the nodes/subobjects themselves. We could add `parse` functions to the nodes/subobjects which take in their unparsed counter part and return the "parsed" version. This doesn't make it less mechnanical, but it does change where the logic is to perhaps somewhere more tasteful.

Yet another way would be what we do for the Pydantic implementations in DSI wherein we put things through the ManifestTransformer. However I'd advocated against this, as you lose some type guarantees as a lot more things become `Optional`.

I could be wrong, but I don't imagine this code being to high touch after 1.6. The current implementation makes it easy to see whats going on and raises discrepancies better than if we add magic, which I think are both important for a future where this code is infrequently touched.